packages/shared/lib/services/sync/config/config.service.ts

```diff
@@ -569,6 +569,54 @@ export async function getSyncConfigsWithConnectionsByEnvironmentId(environment_i
     return result;
 }
 
+export async function getSyncConfigsWithConnectionsByEnvironmentIdAndProviderConfigKey(
+    providerConfigKey: string,
+    environment_id: number
+): Promise<({ connections: { connection_id: string }[] } & SyncConfig & ProviderConfig)[]> {
+    const result = await schema()
```

Comment:
Happy to get another pair of eyes on this query to improve it.

Reply:
I guess you only want connection with syncs? I feel like it's harder to explain to customers but whatever :D 
Isn't something like this enough? I'm not sure why you selected that much stuff from the table

```sql
SELECT
	COUNT(1)
FROM
	_nango_connections
JOIN _nango_syncs ON _nango_syncs.nango_connection_id = _nango_connections.id AND _nango_syncs.deleted = FALSE
WHERE
	environment_id = 2
	AND provider_config_key = 'github-demo'
	AND _nango_syncs.deleted = FALSE;
```

and if it's only 3 connections ever
```sql
SELECT
	COUNT(1)
FROM
	_nango_connections
WHERE
	environment_id = 2
	AND provider_config_key = 'github-demo'
	AND _nango_syncs.deleted = FALSE;
```

Reply:
This give you all connections with active syncs or actions for a provider.
```sql 
SELECT
	conn.id
FROM
	_nango_configs AS configs
	JOIN _nango_connections AS conn ON configs.id = conn.config_id
		AND conn.deleted = FALSE
	JOIN _nango_sync_configs AS syncs ON syncs.nango_config_id = configs.id
		AND syncs.deleted = FALSE
		AND syncs.active = TRUE
WHERE
	configs.environment_id = 2
        AND configs.provider_config_key = 'github-demo'
	AND configs.deleted = FALSE
GROUP BY
	conn.id
HAVING
	COUNT(syncs.id) > 0;
```

then `res.length >= 3`

Reply:
Thanks, this doesn't give me what I want either 

```
SELECT
	conn.id
FROM
	nango._nango_configs AS configs
	JOIN nango._nango_connections AS conn ON configs.id = conn.config_id
		AND conn.deleted = FALSE
	JOIN nango._nango_sync_configs AS syncs ON syncs.nango_config_id = configs.id
		AND syncs.deleted = FALSE
		AND syncs.active = TRUE
WHERE
	configs.environment_id = 2
        AND configs.unique_key = 'unauthenticated'
	AND configs.deleted = FALSE
GROUP BY
	conn.id
HAVING
	COUNT(syncs.id) > 0;
```

gives me one row 
![image](https://github.com/NangoHQ/nango/assets/1724137/1f10f0c3-6f08-459c-b9d3-93ee2a11b646)

![image](https://github.com/NangoHQ/nango/assets/1724137/dabde311-4402-4828-adda-1f40d82f0b34)

Reply:
Yeah indeed, I realized that `_nango_sync_configs` are per config not per connection and `_nango_syncs` are per connection but do not contain actions :/ 
I guess you got the right one then!